### PR TITLE
Adds support for user-supplied metadata

### PR DESCRIPTION
**Pull request by dbernstein**
Resolves: https://jira.duraspace.org/browse/FCREPO-2352

### CODE REVIEW

File: src/main/java/org/fcrepo/importexport/exporter/Exporter.java [hunk 1]
```diff
@@ -129,6 +135,8 @@ public Exporter(final Config config, final FcrepoClient.FcrepoClientBuilder clie
                 bagdir.mkdirs();
                 this.bag = new Bag(new Version(0, 97));
                 this.bag.setRootDir(bagdir);
+                this.bag.getMetadata().putAll(bagConfig.getBagInfo());
+                //TODO where to add aptrust bag info from the bagConfig?
```
Comment (escowles) [inline src/main/java/org/fcrepo/importexport/exporter/Exporter.java]: I don't know if there is support for custom tag files in bagit-java — I looked at the API docs and don't see it anywhere.  So we may have to write our own.  Luckily, the format is just

```
key: value
```

So it shouldn't be hard to just iterate over the APTrust values and output them.

Comment (dbernstein) [inline src/main/java/org/fcrepo/importexport/exporter/Exporter.java]: @escowles :  so are you suggesting that we just write in our own aptrust-info.txt filling it with kv pairs from the bagConfig.getAPTrustInfo()?

File: src/test/resources/configs/bagit-config.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  Contact-Name: My Name
+  Contact-Phone: +19999999999
+  Contact-Email: anyone@university.edu
+  External-Description: ???
+  External-Identifier: ???
+  Bag-Group-Identifier: ???
+  Internal-Sender-Identifier: ???
+  Internal-Sender-Description: ???
+aptrust-info.txt:
+  Access: something
```
Comment (dbernstein) [inline src/test/resources/configs/bagit-config.yml]: will do.

File: src/test/java/org/fcrepo/importexport/common/BagConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+public class BagConfigTest {
+
+    @Test
+    public void testFromFile() throws Exception {
+        final File testFile = new File("src/test/resources/configs/bagit-config.yml");
+        final BagConfig config = new BagConfig(testFile);
+
+        final Map<String,String> bagInfo = config.getBagInfo();
+        assertNotNull(bagInfo);
+        assertNotNull(bagInfo.get(BagConfig.SOURCE_ORGANIZATION_KEY));
```
Comment (dbernstein) [inline src/test/java/org/fcrepo/importexport/common/BagConfigTest.java]: will do.

File: src/main/java/org/fcrepo/importexport/exporter/Exporter.java (hunk 1 above)
Comment (escowles) [inline src/main/java/org/fcrepo/importexport/exporter/Exporter.java]: Yes, for reference, that's what BagWriter is doing: https://github.com/LibraryOfCongress/bagit-java/blob/master/src/main/java/gov/loc/repository/bagit/writer/BagWriter.java#L255

File: src/main/java/org/fcrepo/importexport/exporter/Exporter.java [hunk 4]
```diff
@@ -129,6 +135,8 @@ public Exporter(final Config config, final FcrepoClient.FcrepoClientBuilder clie
                 bagdir.mkdirs();
                 this.bag = new Bag(new Version(0, 97));
                 this.bag.setRootDir(bagdir);
+                this.bag.getMetadata().putAll(bagConfig.getBagInfo());
```
Comment (escowles) [inline src/main/java/org/fcrepo/importexport/exporter/Exporter.java]: @dbernstein: I've created a new PR that adds a fix for this to your commits: https://github.com/fcrepo4-labs/fcrepo-import-export/pull/66

**Pull request merged**